src/Symfony/Component/HttpKernel/DependencyInjection/AddCacheWarmerPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\PriorityTaggedServiceTrait;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+
+/**
+ * Registers the cache warmers.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ */
+class AddCacheWarmerPass implements CompilerPassInterface
+{
+    use PriorityTaggedServiceTrait;
+
+    private $tag;
+    private $serviceName;
+
+    public function __construct($serviceName = 'cache_warmer', $tag = 'kernel.cache_warmer')
+    {
+        if (2 > func_num_args()) {
```

Comment:
> but I don't think it is really useful,

This is part of component now.. imo. it shoud make no assumptions on naming, hence required args :+1: 

>  So you can still continue to use with new ... and without args

Only until 4.0, as we already decided that. Im not sure why we want to bridge that with optional args now (again; for a _new_ class (definition)). It doesnt make sense to me nor does it solve a DX issue, as the old class still works without args.